auth0/src/main/java/com/auth0/android/provider/LogoutManager.java

```diff
@@ -0,0 +1,87 @@
+package com.auth0.android.provider;
+
+import android.content.Context;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.auth0.android.Auth0;
+import com.auth0.android.Auth0Exception;
+import com.auth0.android.callback.BaseCallback;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@SuppressWarnings("WeakerAccess")
+class LogoutManager extends ResumableManager {
+
+    private static final String TAG = LogoutManager.class.getSimpleName();
+
+    private static final String KEY_CLIENT_ID = "client_id";
+    private static final String KEY_TELEMETRY = "auth0Client";
+
+    private final Auth0 account;
+    private final BaseCallback<Void, Auth0Exception> callback;
+    private final Map<String, String> parameters;
+
+    private CustomTabsOptions ctOptions;
+
+    LogoutManager(@NonNull Auth0 account, @NonNull BaseCallback<Void, Auth0Exception> callback, @NonNull Map<String, String> parameters) {
+        this.account = account;
+        this.callback = callback;
+        this.parameters = new HashMap<>(parameters);
+    }
+
+    void setCustomTabsOptions(@Nullable CustomTabsOptions options) {
+        this.ctOptions = options;
+    }
+
+    void startLogout(Context context) {
+        addClientParameters(parameters);
+        Uri uri = buildLogoutUri();
+
+        AuthenticationActivity.authenticateUsingBrowser(context, uri, ctOptions);
+    }
+
+    @Override
+    boolean resume(AuthorizeResult result) {
+        if (result.isCanceled()) {
+            Auth0Exception exception = new Auth0Exception("The user closed the browser app and the log out was cancelled.");
```

Comment:
Is the user closing the browser app the only thing that could trigger this? If so looks good, though maybe change "log out" to "logout".

Reply:
@joshcanhelp This is an exception passed to the callback. Dev decides what to do when they receive the `onFailure` call with this error instance: e.g. if they want to display the very same message or display something different, like a retry dialog.

@jimmyjames  Now that `federated` support was removed, the only scenario here is when the user closes the browser manually. This can also be triggered if the log out URL is not whitelisted in the application or tenant settings. Because the user would need to manually exit that page, that would count as a "cancel" but TBH, this is dev error and should be catch before shipping the app.

Reply:
For the exception text:

> The user closed the browser app so the logout was cancelled.